packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -391,11 +391,30 @@ export type SdkModelPropertyType =
   | SdkBodyParameter
   | SdkHeaderParameter;
 
+export interface MultipartOptionsType {
+  // whether name is defined in Typespec. For multipart/mixed, name may not be defined for some parts
+  isNameDefined: boolean;
```

Comment:
If we haven't decided what to do with the case of no name, we may not want to expose this now.

E.g. we may want to do
```
name = <generated-name>
generatedName = true
serializedName = ""
```
in property. And this be `serializedNameDefined`.

Reply:
This PR doesn't contain case that name is null so I will remove this property for now.